cl/scrapers/management/commands/clone_from_cl.py

```diff
@@ -0,0 +1,406 @@
+dockets_endpoint = "https://www.courtlistener.com/api/rest/v3/dockets/"
+people_endpoint = "https://www.courtlistener.com/api/rest/v3/people/"
+courts_endpoint = "https://www.courtlistener.com/api/rest/v3/courts/"
+
+
+def clone_opinion_cluster(
+    session: Session, cluster_ids: list, object_type="search.OpinionCluster"
+):
+    """
+    Download opinion cluster data from courtlistener.com and add it to
+    local environment
+    :param session: a Requests session
+    :param cluster_ids: a list of opinion cluster ids
+    :param object_type: OpinionCluster app name with model name
+    :return: list of opinion cluster objects
+    """
+
+    opinion_clusters = []
+
+    for cluster_id in cluster_ids:
+        print(f">> Cloning opinion cluster id: {cluster_id}")
+        model = apps.get_model(object_type)
+
+        try:
+            opinion_cluster = model.objects.get(pk=int(cluster_id))
+            print(
+                ">> Opinion cluster already exists here:",
+                reverse(
+                    "view_case",
+                    args=[opinion_cluster.pk, opinion_cluster.docket.slug],
+                ),
+            )
+            opinion_clusters.append(opinion_cluster)
+            continue
+        except model.DoesNotExist:
+            pass
+
+        cluster_url = f"{cluster_endpoint}{cluster_id}/"
+        cluster_datum = session.get(cluster_url, timeout=60).json()
+        docket_id = cluster_datum["docket"].split("/")[7]
+        docket = clone_docket(session, [docket_id])[0]
+        citation_data = cluster_datum["citations"]
+        sub_opinions_data = cluster_datum["sub_opinions"]
+        # delete unneeded fields
+        for f in [
+            "resource_uri",
+            "docket",
+            "citations",
+            "sub_opinions",
+            "absolute_url",
+            "panel",
+            "non_participating_judges",
+        ]:
+            del cluster_datum[f]
+
+        # Assign docket pk in cluster data
+        cluster_datum["docket_id"] = docket.pk
+
+        prepared_opinion_data = []
+        added_opinions_ids = []
+
+        for op in sub_opinions_data:
+            # Get opinion from api
+            op_data = session.get(op, timeout=60).json()
+            # Delete fields with fk or m2m relations or unneeded fields
+            for f in [
+                "opinions_cited",
+                "cluster",
+                "absolute_url",
+                "resource_uri",
+                "author",
+                "joined_by",
+            ]:
+                del op_data[f]
+            # Append new data
+            prepared_opinion_data.append(op_data)
+
+        with transaction.atomic():
+            # Create opinion cluster
+            opinion_cluster = model.objects.create(**cluster_datum)
+
+            for cite_data in citation_data:
+                # Create citations
+                cite_data["cluster_id"] = opinion_cluster.pk
+                Citation.objects.create(**cite_data)
```

Comment:
You could do this with `bulk_create` for a performance boost, but not worth re-doing now. Something to think about in the future though.